app/controllers/posts_controller.rb

```diff
@@ -8,9 +8,30 @@ def index
 
   def show
     @post = Post.find(params[:id])
-    @user = User.find(params[:user_id])
     @post_comments = Comment.where(post_id: @post)
-    @likes = @post.likes
-    @user_posts = @user.posts
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```

Comment:
[OPTIONAL] For future projects I think you should read up on counter cache. It would make your life easier